opencensus_shim/src/main/java/io/opentelemetry/opencensusshim/OpenTelemetryCtx.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.opencensusshim;
+
+import io.opencensus.trace.ContextHandle;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+
+public class OpenTelemetryCtx implements ContextHandle {
+
+  private final Context context;
+  private Scope scope;
```

Comment:
I'm somewhat nervous about holding on to a Scope reference like this. Can we guarantee that attach and detach are always called on the same thread, and no other detach calls will be made on that thread in between the attach/detach pairs? I don't know enough (or anything) about Census to answer this question, but it makes me nervous.

Reply:
FYI the attach/detach methods are an abstraction for gRPC context used in OpenCensus by default:
- https://grpc.github.io/grpc-java/javadoc/io/grpc/Context.html#attach--
- https://grpc.github.io/grpc-java/javadoc/io/grpc/Context.html#detach-io.grpc.Context-

> This context should be the same context that was previously attached. The provided replacement should be what was returned by the same attach() call. If an attach() and a detach() meet above requirements, they match.

> It is expected that between any pair of matching attach() and detach(), all attach()es and detach()es are called in matching pairs. If this method finds that this context is not current, either you or some code in-between are not detaching correctly, and a SEVERE message will be logged but the context to attach will still be bound. Never use Context.current().detach(), as this will compromise this error-detecting mechanism.

Are there any safeguards we can introduce to make sure we have the same semantics in this shim implementation?